src/trace/listener.ts

```diff
@@ -131,8 +133,14 @@ export class TraceListener {
 
     this.lambdaSpanParentContext = this.inferredSpan?.span || parentSpanContext;
     this.context = context;
-    this.triggerTags = extractTriggerTags(event, context);
+    const eventSource = parseEventSource(event);
+    this.triggerTags = extractTriggerTags(event, context, eventSource);
     this.stepFunctionContext = StepFunctionContextService.instance().context;
+
+    const result = getSpanPointerAttributes(eventSource, event);
+    if (result) {
+      this.spanPointerAttributesList = result;
```

Comment:
If this same listener instance is used each time, then I'm worried about the case where the function receives an S3 event (which gets a span pointer) then on the next invocation gets something else. Because we're doing `if (result)` the span pointers will still be present from the previous invocation. Please correct me if I'm wrong though! I think it's probably just better all around to not do the falsy check on the `result`. That's what the step function call a line above is doing.

Reply:
Yes you're correct 🤦. Thank you for catching this